overlord/snapstate/component.go

```diff
@@ -209,17 +239,28 @@ func doInstallComponent(st *state.State, snapst *SnapState, compSetup *Component
 		discardComp := st.NewTask("discard-component", fmt.Sprintf(i18n.G(
 			"Discard previous revision for component %q"),
 			compSi.Component))
+		componentTS.linkToHook = append(componentTS.linkToHook, discardComp)
 		addTask(discardComp)
 	}
 
-	installSet := state.NewTaskSet(tasks...)
-	installSet.MarkEdge(prepare, BeginEdge)
-	installSet.MarkEdge(linkSnap, MaybeRebootEdge)
+	var postOpHook *state.Task
+	if !compInstalled {
+		postOpHook = SetupInstallComponentHook(st, snapsup.InstanceName(), compSi.Component.ComponentName)
+	} else {
+		postOpHook = SetupPostRefreshComponentHook(st, snapsup.InstanceName(), compSi.Component.ComponentName)
+	}
+
+	componentTS.postOpHookAndAfter = append(componentTS.postOpHookAndAfter, postOpHook)
+	addTask(postOpHook)
+
+	componentTS.ts.MarkEdge(prepare, BeginEdge)
+	componentTS.ts.MarkEdge(linkSnap, MaybeRebootEdge)
+	componentTS.ts.MarkEdge(postOpHook, PostOpHookEdge)
 
 	// TODO do we need to set restart boundaries here? (probably
```

Comment:
unrelated but seems this needs to become a TODO:COMPS todo? maybe worth checking other TODOs in this file (cc @alfonsosanchezbeato)